### PR TITLE
RTT Disclosure Banner

### PR DESCRIPTION
**Pull request by carocao-msft**
# What
RTT Disclosure Banner

# Why

# How Tested
Storybook

Screenshot 2024-10-24 at 2 27 27 PM

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/src/components/RTTDisclosureBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * @beta
+ * Banner to disclose that RTT is enabled for all participants for the entire duration of the meeting.
+ */
+export const RTTDisclosureBanner = (props: RTTDisclosureBannerProps): JSX.Element => {
+  const localeStrings = useLocale().strings.rttDisclosureBanner;
+  const strings = { ...localeStrings, ...props.strings };
+  const theme = useTheme();
+
+  return (
+    <Stack className={bannerContainerStyles()}>
```
Comment (JamesBurnside) [inline packages/react-components/src/components/RTTDisclosureBanner.tsx:52]: How come this doesn't reuse the notifications component we have?

File: packages/communication-react/review/beta/communication-react.api.md [hunk 2]
```diff
@@ -2393,6 +2393,8 @@ export interface ComponentStrings {
     raiseHandButton: RaiseHandButtonStrings;
     reactionButton: ReactionButtonStrings;
     richTextSendBox: RichTextSendBoxStrings;
+    // (undocumented)
```
Comment (PorterNan) [inline packages/communication-react/review/beta/communication-react.api.md:2396]: Add jsdoc for this

File: packages/communication-react/review/beta/communication-react.api.md [hunk 3]
```diff
@@ -4427,6 +4431,22 @@ export interface RichTextStrings {
...
+// @beta
+export interface RTTDisclosureBannerProps {
+    link?: string;
+    strings?: RTTDisclosureBannerStrings;
+}
+
+// @beta
+export interface RTTDisclosureBannerStrings {
+    // (undocumented)
+    bannerContent?: string;
```
Comment (PorterNan) [inline packages/communication-react/review/beta/communication-react.api.md:4443]: Jsdoc needed for all undocumented properties

File: packages/react-components/src/components/RTTDisclosureBanner.tsx (hunk 1 above)
Comment (PorterNan) [reply packages/react-components/src/components/RTTDisclosureBanner.tsx:52]: Is t his a banner on the top of composite or on top of captionBanner component?

**Review by PorterNan — APPROVED** [review]

File: packages/react-components/src/components/RTTDisclosureBanner.tsx (hunk 1 above)
Comment (carocao-msft) [reply packages/react-components/src/components/RTTDisclosureBanner.tsx:52]: It's a banner inside the captions banner component

Comment (JamesBurnside) [reply packages/react-components/src/components/RTTDisclosureBanner.tsx:52]: I don't understand banner vs notification, why is this not another notification in the notification stack?

Comment (carocao-msft) [reply packages/react-components/src/components/RTTDisclosureBanner.tsx:52]: Screenshot 2024-10-25 at 9 27 08 AM
This is the design, it's not a notification, it's a sticky banner inside captions banner. It's also of a different design versus the notification component, title, icon, and learn more link and box shadows are all different

Comment (carocao-msft) [reply packages/react-components/src/components/RTTDisclosureBanner.tsx:52]: Screenshot 2024-10-25 at 9 31 37 AM

**Review by JamesBurnside — APPROVED** [review]

**Review by PorterNan — APPROVED** [review]

File: packages/communication-react/review/stable/communication-react.api.md [hunk 4]
```diff
@@ -3413,10 +3414,12 @@ export interface NotificationProps {
     autoDismiss?: boolean;
     notificationIconProps?: IIconProps;
     notificationStrings?: NotificationStrings;
+    onClickLink?: () => void;
```
Comment (PorterNan) [inline packages/communication-react/review/stable/communication-react.api.md:3417]: What is this onClickLink doing btw?

Comment (carocao-msft) [reply packages/communication-react/review/stable/communication-react.api.md:3417]: It's fired when the link is clicked.

**Pull request merged**